### DISCUSSION TITLE
trino cli authentication not work

### DISCUSSION THREAD

**Discussion by s-amir** [trinodb/trino#23333] [Q&A]
I enable trino to use ldap to authenticate users , i mean when i connect to trino UI over port 8443 it needs user/pass and check with ldap to authenticate 
but as i want to use trino cli it does not work properly!
with command "trino --server https://127.0.0.1:8443 --user=amirdarvish --password" it ask for password but connect with any user and pass that are not correct!
However i can not run any query because it returns error:
```
javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```
and also there is no log in ldap with connecting with CLI!

here is my config.properties:
```
coordinator=true
node-scheduler.include-coordinator=true
http-server.http.port=8080
catalog.management=${ENV:CATALOG_MANAGEMENT}
http-server.authentication.type=PASSWORD
http-server.http.enabled=false
http-server.https.port=8443
discovery.uri=https://localhost:8443
http-server.https.keystore.path=/etc/trino/certs/trino-combined.pem
internal-communication.shared-secret=supersecretpassword123!
internal-communication.https.required=true
http-server.https.enabled=true



```
password-authenticator.properties:
```
password-authenticator.name=ldap
ldap.url=ldaps://ldapserver:636
ldap.ssl.truststore.path=/etc/trino/certs/ldap-server-crt.pem
#ldap.ssl.truststore.password=changeit
#ldap.user-bind-pattern=CN=%s,DC=txvlab,DC=local
ldap.user-bind-pattern=cn=${USER},cn=admin,dc=txvlab,dc=local

```

Can anyone help me ?

**Comment by hashhar** [comment]
you also need to provide the Trino CLI with the keystore since you are using self-signed certs probably.

See `--keystore-path` on https://trino.io/docs/current/client/cli.html
